source/extensions/filters/http/credential_injector/config.cc

```diff
@@ -31,21 +32,27 @@ CredentialInjectorFilterFactory::createFilterFactoryFromProtoTyped(
   ProtobufTypes::MessagePtr message = Config::Utility::translateAnyToFactoryConfig(
       proto_config.credential().typed_config(), context.messageValidationVisitor(),
       *config_factory);
+
+  // In upstream filter call, prefix has cluster.cluster_0 without '.', needs to be added
+  std::string tmp_prefix = dual_info.is_upstream ? stats_prefix + "." : stats_prefix;
   CredentialInjectorSharedPtr credential_injector =
       config_factory->createCredentialInjectorFromProto(
-          *message, stats_prefix + "credential_injector.", context);
+          *message, tmp_prefix + "credential_injector.", context);
 
-  FilterConfigSharedPtr config =
-      std::make_shared<FilterConfig>(std::move(credential_injector), proto_config.overwrite(),
-                                     proto_config.allow_request_without_credential(),
-                                     stats_prefix + "credential_injector.", context.scope());
+  FilterConfigSharedPtr config = std::make_shared<FilterConfig>(
+      std::move(credential_injector), proto_config.overwrite(),
+      proto_config.allow_request_without_credential(),
+      dual_info.is_upstream ? "credential_injector." : stats_prefix + "credential_injector.",
```

Comment:
this is another side effect, it is called twice. without this change we get cluster.cluster_0.cluster.cluster_0credential_injector